HW04.Rmd

```diff
@@ -1,126 +1,306 @@
+
+kable(new_filter_race, format = "markdown",
+      col.names = c("Race", "Intent", "Number of Deaths", "Percentage of Death (by race)"),
+      caption = "Total number of gun related deaths by Race and Intent")
+
+#create plot
+ggplot(new_filter_race, aes(race, percent, fill = intent)) +
+    geom_col(position = "fill") +
+    
+#add labels
+    labs(title = "Percentage of Gun Deaths by Race from 2012-2014",
+         x     = "Race",
+         y     = "Percentage of Deaths") 
+
+```
+
+
+### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
+
+```{r police}
+#filter and clean data
+gun_deaths_police <- gun_deaths %>%
+    select(age, police, race) %>%
+    filter(age != "Not Available") %>%
+    mutate(age = as.numeric(age))
+
+# generating a boxplot visualizing the data
+ggplot(gun_deaths_police, aes(police, age, group = police)) +
+    geom_boxplot(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 100, by = 10)) +
+    scale_x_discrete(breaks = c("0.0", "1.0"), labels = c("Uninvolved", "Involved")) +
+        #it does not seem to be labeling "Uninvolved vs Involved" on the x-axis and I am unsure why.
+    
+#add labels
+    labs(title = "Gun Related Deaths Involving Police from 2012-2014",
+         x     = "Police Involvement",
+         y     = "Age (years)") +
+
+ #add theme   
+    theme_classic()
+```
+
+Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
+
+```{r stats?}
+# create  variables for t.test(x,y) function
+police_uninvolved_age <- gun_deaths %>%
+    select(police, age) %>%
+    mutate(age = as.numeric(age)) %>%
+    filter(police == 0)
+
+police_involved_age <- gun_deaths %>%
+    select(police, age) %>%
+    mutate(age = as.numeric(age)) %>%
+    filter(police == 1)
+
+# t-test 
+t.test(police_uninvolved_age$age, police_involved_age$age)
+
+```
+
+
+```{r police_2}
+
+```
+
+```{r stats_2?}
+
+
+```
+
+
+Finally, all together now:
+```{r police_3}
+#filter data
+gun_deaths_all_together <- gun_deaths %>%
+    select(age, race, police) %>%
+    mutate(age = as.numeric(age)) %>%
+    group_by(race, police)
+
+#create plot
+ggplot(gun_deaths_all_together, aes(police, age, group = police, fill = police)) + 
+            #i know the continuous scale of police on the side doesn't make much sense but I can't seem to label this how i want using scale_x_discrete
+    geom_boxplot() +
+    facet_wrap(~race, nrow = 1, labeller = label_wrap_gen(width = 20)) +
+
+#add labels
+    scale_y_continuous(breaks = seq(0, 100, by = 10)) +
+    scale_x_discrete(breaks=c("0", "1"), labels=c("Uninvolved", "Involved")) + #still not showing labels when i run this
+
+#add labels
+    labs(title = "Gun Related Deaths from 2012-2014",
+         x     = "Police Involvement",
+         y     = "Age (years)") 
+```
```

Comment:
Visualizing multiple relationships in 2-dimensions can be pretty challenging, but your plot does a good job of this.